source/plugins/go/src/oms.go

```diff
@@ -1357,7 +1357,7 @@ func PostDataHelper(tailPluginRecords []map[interface{}]interface{}) int {
 		numContainerLogRecords = len(dataItemsADX)
 		Log("Success::ADX::Successfully wrote %d container log records to ADX in %s", numContainerLogRecords, elapsed)
 
-	} else { //ODS
+	} else if ((ContainerLogSchemaV2 == true && len(dataItemsLAv2) > 0) || len(dataItemsLAv1) > 0) { //ODS
```

Comment:
can we remove the ods code path ? If you had removed the path in the first place, we wouldn't have encountered this issue. Yes, we will flush an empty array to mdsd, but the behavior will be the same as before without triggering retries.

Reply:
We cant remove ODS code path since this being used for the Windows agent as we dont have Windows AMA integration.  Until we integrate with Windows AMA, we have to keep this code path.

Reply:
sorry, dont agree. we can certainly remove ods code path in linux completely.

Reply:
ODS code path used for windows and for Linux, as backup option, where we can suggest customer configures the "v1" route via config map in case if we ended getting any incidents related to container log missing with mdsd in high container log volume customers.  Certainly, we can remove ods path for linux completly in august or later release. lets keep this backup option as is. we can discuss more if you have concern on this.

Reply:
where did we discuss building a feature like this to revert back to v1 ?

Reply:
> where did we discuss building a feature like this to revert back to v1 ?

> Please refer the add stage2 PR where I have mentioned and also responded to similar comment - https://github.com/microsoft/Docker-Provider/pull/559

Reply:
given we had this bug, what you had in the old PR doesn't matter. Just check that for linux node this path is completely mooted. We don't want a 'hidden' unused feature introducing bugs, just like it did. Previously this was needed, now we dont. Also, for comments in PR that you made no changes, please ensure the person putting the comment agree to your theories. You cannot just resolve them by yourselves, and ask someone else to approve the PR and merge.

Reply:
created a task to address this august agent release to moot the ODS code path completely.